otaclient/app/errors.py

```diff
@@ -88,240 +68,194 @@ class OTAError(Exception):
+    failure_errcode: OTAErrorCode = OTAErrorCode.E_OTAMETA_DOWNLOAD_FAILED
+    failure_description: str = (
+        f"failed to download OTA meta due to {_NETWORK_ERR_DEFAULT_DESC}"
+    )
+
 
+#
+# ------ recoverable error ------
+#
 
-### recoverable error ###
 _RECOVERABLE_DEFAULT_DESC = (
-    "recoverable ota error(unrelated to network) detected, "
-    "please resend the request or retry after a restart of ota-client/ECU"
+    "recoverable OTA error(unrelated to network) detected, "
+    "please retry after reboot device or restart otaclient"
 )
 
 
 class OTAErrorRecoverable(OTAError):
     failure_type: wrapper.FailureType = wrapper.FailureType.RECOVERABLE
-    # followings are default values
-    module: OTAModules = OTAModules.General
-    errcode: OTAErrorCode = OTAErrorCode.E_OTA_ERR_RECOVERABLE
-    desc: str = _RECOVERABLE_DEFAULT_DESC
+    failure_errcode: OTAErrorCode = OTAErrorCode.E_OTA_ERR_RECOVERABLE
+    failure_description: str = _RECOVERABLE_DEFAULT_DESC
 
 
-class OTAUpdateBusy(OTAErrorRecoverable):
-    module: OTAModules = OTAModules.API
-    errcode: OTAErrorCode = OTAErrorCode.E_OTAUPDATE_BUSY
-    desc: str = f"{_RECOVERABLE_DEFAULT_DESC}: on-going ota update detected, this request has been ignored"
+class OTABusy(OTAErrorRecoverable):
+    failure_errcode: OTAErrorCode = OTAErrorCode.E_OTA_BUSY
+    failure_description: str = "on-going OTA operation(update or rollback) detected, this request has been ignored"
 
 
-class OTARollBusy(OTAErrorRecoverable):
-    module: OTAModules = OTAModules.API
-    errcode: OTAErrorCode = OTAErrorCode.E_OTAUPDATE_BUSY
-    desc: str = f"{_RECOVERABLE_DEFAULT_DESC}: on-going ota update detected, this request has been ignored"
-
+class InvalidStatusForOTARollback(OTAErrorRecoverable):
+    failure_errcode: OTAErrorCode = OTAErrorCode.E_INVALID_STATUS_FOR_OTAROLLBACK
+    failure_description: str = "previous OTA is not succeeded, reject OTA rollback"
 
-class InvalidStatusForOTAUpdate(OTAErrorRecoverable):
-    module: OTAModules = OTAModules.API
-    errcode: OTAErrorCode = OTAErrorCode.E_INVALID_STATUS_FOR_OTAUPDATE
-    desc: str = f"{_RECOVERABLE_DEFAULT_DESC}: current ota-status indicates it should not accept ota update"
 
+#
+# ------ recoverable error ------
+#
 
-class InvalidUpdateRequest(OTAErrorRecoverable):
-    module: OTAModules = OTAModules.API
-    errcode: OTAErrorCode = OTAErrorCode.E_INVALID_OTAUPDATE_REQUEST
-    desc: str = (
-        f"{_RECOVERABLE_DEFAULT_DESC}: incoming ota update request's content is invalid"
-    )
+_UNRECOVERABLE_DEFAULT_DESC = (
+    "unrecoverable OTA error, please contact technical support"
+)
 
 
-class InvalidStatusForOTARollback(OTAErrorRecoverable):
-    module: OTAModules = OTAModules.API
-    errcode: OTAErrorCode = OTAErrorCode.E_INVALID_STATUS_FOR_OTAROLLBACK
-    desc: str = f"{_RECOVERABLE_DEFAULT_DESC}: current ota-status indicates it should not accept ota rollback"
+class OTAErrorUnRecoverable(OTAError):
```

Comment:
```suggestion
class OTAErrorUnrecoverable(OTAError):
```
or if we use `UnRecoverable`, then we need to employee `UN_RECOVERABLE`. `Unrecoverable` is natural I think.

Reply:
Updated in https://github.com/tier4/ota-client/pull/259/commits/16cc689223d26e5adce9534acb09df8c64b8cc8d